crates/e2e/src/xts.rs

```diff
@@ -33,43 +33,54 @@ use subxt::{
     blocks::ExtrinsicEvents,
     config::ExtrinsicParams,
     rpc_params,
-    tx,
+    utils::MultiAddress,
     OnlineClient,
 };
 
 /// A raw call to `pallet-contracts`'s `instantiate_with_code`.
-#[derive(Debug, scale::Encode, scale::Decode)]
-pub struct InstantiateWithCode<B> {
+#[derive(Debug, scale::Encode, scale::Decode, scale_encode::EncodeAsType)]
+#[encode_as_type(trait_bounds = "")]
+pub struct InstantiateWithCode<E: Environment> {
     #[codec(compact)]
-    value: B,
-    gas_limit: Weight,
-    storage_deposit_limit: Option<B>,
+    value: E::Balance,
+    gas_limit: subxt::utils::Static<Weight>,
+    storage_deposit_limit: Option<E::Balance>,
     code: Vec<u8>,
     data: Vec<u8>,
     salt: Vec<u8>,
 }
 
 /// A raw call to `pallet-contracts`'s `call`.
-#[derive(Debug, scale::Encode, scale::Decode)]
-pub struct Call<E: Environment, B> {
-    dest: sp_runtime::MultiAddress<E::AccountId, ()>,
+#[derive(Debug, scale::Decode, scale::Encode, scale_encode::EncodeAsType)]
+#[encode_as_type(trait_bounds = "")]
+pub struct Call<E: Environment> {
+    dest: MultiAddress<E::AccountId, ()>,
     #[codec(compact)]
-    value: B,
-    gas_limit: Weight,
-    storage_deposit_limit: Option<B>,
+    value: E::Balance,
+    gas_limit: subxt::utils::Static<Weight>,
```

Comment:
`Weight` is a super simple struct, so for a little added robustness/simplicity I wonder whether it's worth just writing your own version that impls EncodeAsType et al

Reply:
I'm fine with a local copy of the type with a minimal API, it is likely to be highly stable.

Reply:
done